src/search-box/search-box.ts

```diff
@@ -131,6 +134,19 @@ export class NgAisSearchBox
     }
   }
 
+  public ngDoCheck() {
+    // We bypass the state update if the input is focused to avoid concurrent
+    // updates when typing.
+    if (
+      this.searchBox &&
+      this.searchBox.nativeElement &&
+      document.activeElement !== this.searchBox.nativeElement &&
+      this.query !== this.state.query
```

Comment:
I think it makes a bit more sense to have this check first, as this is what we're tracking initially.